joomla/joomla-cms
# [#31516] Start a JLayout Social

This starts the process of making some features not already in the CMS available as JLayouts. This starts with:
- Facebook Like
- Twitter Share
- Twitter Follow
- Twitter Hashtag
- Twitter Mention
- Google +1

Tracker: http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=31516
## Screenshot

![image](https://f.cloud.github.com/assets/1986000/801702/e531a0c8-eda8-11e2-8032-e401b9481fcc.png)

Note including them in articles is **_NOT**_ included in this PR - this is just demonstrating their use
## Layout Params
### Google+1
- **data-size** _small, medium, standard, tall_ Size of the icon - default standard.
- **data-annotation** _inline, bubble, none_ The style of annotation to appear - default inline.
- **data-width** _integer_ The width of the icon - default 300
- **data-href** _URL_ The URL for the like to link to. By default the current URL (JUri::current())
- **show-count** _true/false_ Show/Hide the number of people who have +1'd the page
- **language** _Language string recognized by google_ Defaults to the locale according to the language (https://github.com/joomla/joomla-cms/blob/master/libraries/joomla/language/language.php#L1213).
### Facebook Like
- **appid** _integer_ The appid if one exists for the like to link with
- **data-href** _URL_ The URL for the like to link to. By default the current URL (JUri::current())
- **data-width** _integer_ The width of the icon - default 450
- **show-faces** _true/false_ Shows the faces of others who have liked the page. Default true.
- **data-layout** _standard, button_count, box_count_ Layout of button. Default standard.
- **data-action** _'empty attribute'/like, recommend_ Verb for the button. Default empty (which is "like").
- **data-colorscheme** _light, dark_ Color scheme. Default like
- **language** _Language string recognized by fb_ Defaults to the locale according to the language (https://github.com/joomla/joomla-cms/blob/master/libraries/joomla/language/language.php#L1213).
### Twitter
#### Follow
- **show-count** _true/false_ Show the number of users followed. Default false
- **user** _Valid Twitter user_ The user for twitter. If not entered will give a JLog priority warning.
#### Share

To be written up (see code for now)
#### Hashtag

To be written up (see code for now)
#### Mention

To be written up (see code for now)
## Issues
- Is my new language file in a good location and with a good name?
- Twitter hashtag langs don't seem to work at the moment (the other twitter views do)
- Is the location of the social layouts package in a good place?
## Testers
- Check multiple instances of each icon work
- Check each icon work
- Check parameters work as expected
- Check multilingual aspect

Suggested code:

```
$layout = new JLayoutFile('social.google');
echo $layout->render(array());
$layout2 = new JLayoutFile('social.twitter.share');
echo $layout2->render(array());
$layout3 = new JLayoutFile('social.facebook');
echo $layout3->render(array());
$layout4 = new JLayoutFile('social.twitter.follow;);
echo $layout4->render(array('user'=>'GW1992'));  // or similar for a twitter username (please don't spam mine :P)
```
